Source/Controller/RawMessageCoordinator.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  RawMessageCoordinator.swift
+//  Planetary
+//
+//  Created by Martin Dutra on 19/9/22.
+//  Copyright © 2022 Verse Communications Inc. All rights reserved.
+//
+
+import Foundation
+import Logger
+
+/// A coordinator for the `RawMessageView`
+@MainActor class RawMessageCoordinator: RawMessageViewModel {
+
+    private var keyValue: KeyValue
+
+    private var bot: Bot
+
+    @Published var rawMessage: String?
+    
+    @Published var loadingMessage: String?
+
+    @Published var errorMessage: String?
+
+    init(keyValue: KeyValue, bot: Bot) {
+        self.keyValue = keyValue
+        self.bot = bot
+        loadRawMessage()
+    }
+
+    func didDismiss() {
+        AppController.shared.dismiss(animated: true)
+    }
+
+    func didDismissError() {
+        errorMessage = nil
+        didDismiss()
+    }
+
+    private func loadRawMessage() {
+        loadingMessage = Text.loading.text
+        Task {
```

Comment:
Probably it's better to detach this task so that it doesn't run on the MainActor (and I realize you were probably copying my code from one of the recent coordinators I wrote 😬 )
```suggestion
        Task.detached {
```

Reply:
I had to add two functions to update the coordinator's variable from the main actor. I had two choices, etiher
```
func updateRawMessage(_ rawMessage: String) {
   self.rawMessage = rawMessage
}
await updateRawMessage(rawMessage)
```
 or 
```
await MainActor.run { [weak self, rawMessage] in
   self?.rawMessage = rawMessage
}
```
I chose the first one